src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -273,7 +275,18 @@ def execute(self, emulator=None):
 
         os.environ.update(envOverride)
 
-        returnCode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        startTime = time.time()
+        # returnCode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        # proc = psutil.Popen(args, stdout=stdoutHandle, stderr=stderrHandle)
+        proc = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        childPid, exitStatus, resource = os.wait4(proc.pid, os.P_PID)
+        stdout, stderr = proc.communicate()
+        stdoutHandle.write(str(stdout))
+        stderrHandle.write(str(stderr))
+        returnCode = proc.returncode
+        userTime = getrusage(RUSAGE_CHILDREN).ru_utime
+        sysTime = getrusage(RUSAGE_CHILDREN).ru_stime
```

Comment:
As far as I've understood, the `getrusage()` returns the resources the process (tree) has consumed up to the call point. Therefore, in order to record only the resources consumed by the subprocess launched by the `Popen()` call, the user and system time would have to be recorded both before and after the call to `Popen()`, and their difference would be the ones stored to the `userTime` and `sysTime` member variables.

Reply:
Matti, good catch, I think you are right here. And I adjusted the code accordingly, see this commit https://github.com/dmwm/WMCore/pull/11665/commits/c70cedd8ab7c716d30f583c1aef4e17c863272b2

Reply:
Looks good by eye, thanks!

Reply:
I'll try to add appropriate unit test soon to measure the difference of times of main code flaw and with subprocess call

Reply:
Matti, have a look at new unit test in this commit: https://github.com/dmwm/WMCore/pull/11665/commits/9216daeb08f56982b40bf6f22eafae9a20e31d08 where I tested logic of getting subprocess time based on your observation.